pandas/core/frame.py

```diff
@@ -2933,12 +2933,12 @@ def __setitem__(self, key, value):
             # set column
             self._set_item(key, value)
 
-    def _setitem_slice(self, key, value):
+    def _setitem_slice(self, key: slice, value):
         # NB: we can't just use self.loc[key] = value because that
         #  operates on labels and we need to operate positional for
         #  backwards-compat, xref GH#31469
         self._check_setitem_copy()
-        self.loc._setitem_with_indexer(key, value)
+        self.iloc[key] = value
```

Comment:
`iloc` ends up calling the same method, so calling it here directly probably avoids a bunch of duplicate validation. 

So I would either leave it as is, or at least check the impact of that, or eg change to `iloc._setitem_with_indexer(..)` to make it more obvious that it is positional indexing

Reply:
WIll do that for now and revisit in a non-"cleanup" PR

Reply:
updated